app/questionnaire/navigator.py

```diff
@@ -1,193 +1,293 @@
 import logging
 
+from app.data_model.answer_store import AnswerStore
+
 logger = logging.getLogger(__name__)
 
 
-def evaluate_rule(rule, answer):
+def evaluate_rule(rule, answer_value):
     """
     Determine whether a rule will be satisfied based on a given answer
     :param rule:
-    :param answer:
+    :param answer_value:
     :return:
     """
     when = rule['when']
     match_value = when['value']
     condition = when['condition']
 
     # Evaluate the condition on the routing rule
-    if condition == 'equals' and match_value == answer:
+    if condition == 'equals' and match_value == answer_value:
         return True
-    elif condition == 'not equals' and match_value != answer:
+    elif condition == 'not equals' and match_value != answer_value:
         return True
     return False
 
 
-def build_path(blocks, block_id, answers, path):
+def evaluate_goto(goto_rule, answers, group_instance):
     """
-    Recursive method which visits all the blocks and returns path taken
-    given a list of answers
-
-    :param blocks: A list containing all blocks in the survey
-    :param block_id: The block id to visit
-    :param answers: The answers to use on evaluating rules
-    :param path: The known path as a list which has been visited already
-    :return: A list of block ids followed through the survey
+    Determine whether a goto rule will be satisfied based on a given answer
+    :param goto_rule:
+    :param answers:
+    :param group_instance:
+    :return:
     """
-    if block_id in Navigator.CLOSING_INTERSTITIAL_PATH:
-        return path
+    if 'when' in goto_rule.keys():
+        answer_index = goto_rule['when']['id']
+        filtered = answers.filter(answer_id=answer_index, group_instance=group_instance)
+        if len(filtered) == 1:
+            answer = filtered[0]
+            if evaluate_rule(goto_rule, answer['value']):
+                return True
+        else:
+            return False
+    elif 'id' in goto_rule.keys():
+        return True
+    return None
```

Comment:
It would be better if we only had true false return values for a boolean evaluation